### ISSUE TITLE
Build steps should be retry-able in case of an agent disconnect

### ISSUE THREAD

**Issue by rafikk** [buildkite/feedback#108]
Presently, if the agent running a build step disconnects for any reason, Buildkite marks the step as failed. This is a problem for users who run build agents in elastic environments where agent nodes may come and go.

**Comment by therealbnut** [comment]
:+1:

**Comment by keithpitt** [comment]
Yup, we should look into adding an option for this! Thanks for the idea!

**Comment by irfansharif** [comment]
I have a use case similar to what Rafik mentioned earlier, running builds on servers working in parallel spontaneously spun up and down. I'd prefer functionality where the current step in the build process is put back on a queue of sorts for other nodes to work on if the step was interrupted mid-process.

**Comment by keithpitt** [comment]
@irfansharif yup! That's exactly what we'll end up creating I think!

**Comment by petemounce** [comment]
Ah, this would aid our autoscaling ideas (cc @buchgr)
